tests/benchmark.py

```diff
@@ -91,29 +91,44 @@ def benchmark_dhfr():
 
     lamb = 0.0
 
-    start = time.time()
-
     writer = PDBWriter([host_pdb.topology], "dhfr.pdb")
 
     num_batches = 100
     steps_per_batch = 1000
+    seconds_per_day = 86400
+    batch_times = []
+
+    lambda_schedule = np.ones(steps_per_batch)*lamb
+
+    # run once before timer starts
+    ctxt.multiple_steps(lambda_schedule)
+
+    start = time.time()
 
     for batch in range(num_batches):
-        lambda_schedule = np.ones(steps_per_batch)*lamb
+
+        # time the current batch
+        batch_start = time.time()
         ctxt.multiple_steps(lambda_schedule)
+        batch_end = time.time()
+
+        delta = batch_end - batch_start
+
+        batch_times.append(delta)
 
-        delta = time.time()-start
-        steps_per_second = (batch+1)*steps_per_batch/delta
-        seconds_per_day = 86400
+        steps_per_second = steps_per_batch / np.mean(batch_times)
```

Comment:
Interesting way of computing steps_per_second, seems right to me. note that mean takes O(N) take, but in practice this doesn't matter.